spikeinterface/qualitymetrics/misc_metrics.py

```diff
@@ -326,3 +401,34 @@ def compute_amplitudes_cutoff(waveform_extractor, peak_sign='neg',
         all_fraction_missing[unit_id] = fraction_missing
 
     return all_fraction_missing
+
+
+if HAVE_NUMBA:
+    @numba.jit((numba.int64[::1], numba.int32), nopython=True, nogil=True, cache=True)
+    def _compute_nb_violations_numba(spike_train, t_r):
+        n_v = 0
+        N = len(spike_train)
+
+        for i in range(N):
+            for j in range(i+1, N):
+                diff = spike_train[j] - spike_train[i]
+
+                if diff > t_r:
+                    break
+
+                # if diff < t_c:
+                #     continue
+
+                n_v += 1
+
+        return n_v
+
+    @numba.jit((numba.int32[::1], numba.int64[::1], numba.int32[::1], numba.int32, numba.int32),
+               nopython=True, nogil=True, cache=True, parallel=True)
+    def _compute_rp_violations_numba(nb_rp_violations, spike_trains, spike_clusters, t_c, t_r):
+        n_units = len(nb_rp_violations)
+
+        for i in numba.prange(n_units):
+            spike_train = spike_trains[spike_clusters == i]
+            n_v = _compute_nb_violations_numba(spike_train, t_r)
+            nb_rp_violations[i] += n_v
```

Comment:
Why having 2 function here ? You could easily make one function with nested loop no ?
This would avoid internally call to function at C level.